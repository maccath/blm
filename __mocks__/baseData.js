const baseData = {
  site: {
    siteMetadata: {
      author: "Site Author",
      description: "Site description",
      title: "Site Title",
    },
  },
  placeholderImage: {
    childImageSharp: {
      fluid: {
        src: "icon.png",
        srcSet: "icon.png 50w, icon.png 100w",
        sizes: "(max-width: 200px) 100vw, 200px",
        base64: "data:image/png;base64,iVBORw0KG",
        aspectRatio: 1,
      },
    },
  },
}

export default baseData
