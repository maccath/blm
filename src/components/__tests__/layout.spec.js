import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from "gatsby"
import "jest-styled-components"
import baseData from "../../../__mocks__/baseData"

import Layout from "../layout"

beforeEach(() => {
  useStaticQuery.mockImplementation(() => baseData)
})

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Layout children="Some content" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
