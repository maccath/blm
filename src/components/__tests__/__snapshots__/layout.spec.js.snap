// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  style={
    Object {
      "opacity": 1,
      "transition": "opacity 500ms ease-in-out",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0 1.0875rem 1.45rem",
      }
    }
  >
    <main>
      Some content
    </main>
    <footer
      className="css-1bxtzlt"
    >
      <p>
        Compiled by 
        <a
          href="http://katy-ereira.co.uk"
        >
          Katy Ereira
        </a>
         /
         
        <a
          href="http://twitter.com/maccath"
        >
          @maccath
        </a>
         | Fist icon by
         
        <a
          href="https://thenounproject.com/cesarreynoso562/"
        >
          Cesar Reynoso
        </a>
      </p>
    </footer>
  </div>
</div>
`;
